File: 0997-Find_the_Town_Judge/Find_the_Town_Judge.java
class Solution {
    public int findJudge(int n, int[][] trust) {
        int[] numOfTrust = new int[n + 1];
        int[] numOfTrusted = new int[n + 1];
        
        for(int i = 0; i < trust.length; i++){
            numOfTrusted[trust[i][1]]++;
            numOfTrust[trust[i][0]]++;
        }
        

        for(int i = 1; i <= n; i++){
            if(numOfTrusted[i] == (n - 1) && numOfTrust[i] == 0){
                return i;
            }
        }
        
        return -1;
    }
}

File: 0997-FindTheTownJudge/FindTheTownJudge.go
package findthetownjudge

func findJudge(n int, trust [][]int) int {
	trusting := make([]int, n+1)
	trusted := make([]int, n+1)

	for i := range trust {
		trusting[trust[i][0]]++
		trusted[trust[i][1]]++
	}

	for i := 1; i <= n; i++ {
		if trusting[i] == 0 && trusted[i] == n-1 {
			return i
		}
	}

	return -1
}
